multi-transfer-esdt/src/lib.rs

```diff
@@ -2,20 +2,24 @@
 
 elrond_wasm::imports!();
 
-use transaction::{esdt_safe_batch::TxBatchSplitInFields, EthTransaction, Transaction};
+use transaction::{
+    esdt_safe_batch::TxBatchSplitInFields, EthTransaction, PaymentsVec, Transaction,
+};
 
 const DEFAULT_MAX_TX_BATCH_SIZE: usize = 10;
 const DEFAULT_MAX_TX_BATCH_BLOCK_DURATION: u64 = u64::MAX;
 
 #[elrond_wasm::contract]
 pub trait MultiTransferEsdt: tx_batch_module::TxBatchModule {
     #[init]
-    fn init(&self) {
+    fn init(&self, #[var_args] opt_wrapping_contract_address: OptionalValue<ManagedAddress>) {
```

Comment:
we have to be prepared to have multiple wrapping contract addresses.

Reply:
Having multiple wrapping contracts makes this SC very heavy. It would need to know to which contract to send. I would much rather extend the current wrapping contract to support pairs of 1 <-> N wrapping/unwrapping.

Reply:
That is good for me as well. But we need something like that.